global.go

```diff
@@ -183,52 +210,109 @@ func globalWorker(task int, m GlobalMethod, g *globalStatus, loc *Location, x []
 	}
 }
 
-// globalOperation updates handles the status received by an individual worker.
-// It uses a mutex to protect updates where necessary.
-func (g *globalStatus) globalOperation(op Operation, loc *Location, x []float64) Status {
-	// Do a quick check to see if one of the other workers converged in the meantime.
+// globalStatus is a data struct that coordinates information between concurrently
+// executing optimization tasks.
+type globalStatus struct {
+	mux       *sync.RWMutex
+	stats     *Stats
+	status    Status
+	p         *Problem
+	startTime time.Time
+	optLoc    *Location
+	settings  *Settings
+	method    GlobalMethod
+	statuser  Statuser
+	err       error
+}
+
+// getStatus returns the current status of the optimization.
+func (g *globalStatus) getStatus() Status {
 	var status Status
-	var err error
 	g.mux.RLock()
+	defer g.mux.RUnlock()
 	status = g.status
-	g.mux.RUnlock()
+	return status
+}
+
+func (g *globalStatus) incrementMajorIteration() {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	g.stats.MajorIterations++
+}
+
+func (g *globalStatus) updateOptLoc(loc *Location) {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	copyLocation(g.optLoc, loc)
+}
+
+// checkConvergence checks the convergence of the global optimization and returns
+// the status
+func (g *globalStatus) checkConvergence() Status {
+	g.mux.RLock()
+	defer g.mux.RUnlock()
+	return checkConvergence(g.optLoc, g.settings, false)
+}
+
+// updateStats updates the evaluation statistics for the given operation.
+func (g *globalStatus) updateStats(op Operation) {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	updateEvaluationStats(g.stats, op)
+}
+
+// updateStatus updates the status and error fields of g. This update only happens
+// if status == NotTerminated, so that the first different status is the one
+// maintained
+func (g *globalStatus) updateStatus(s Status, err error) {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	if g.status != NotTerminated {
+		g.status = s
+		g.err = err
+	}
+}
+
+func (g *globalStatus) finishIteration(status Status, err error, loc *Location, op Operation) (Status, error) {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	return finishIteration(status, err, g.stats, g.settings, g.statuser, g.startTime, loc, op)
+}
+
+// globalOperation executes the requested operation at the given location.
+// Modifications to this function must keep in mind that it can be called
```

Comment:
`"Modifications to this function must keep in mind that it..."`

this reads a bit funny.

perhaps:
`Modifications to this function must be applied while keeping in mind that globalOperation can be...`